### PR TITLE
Thumbnail Image Options

### LINKED ISSUE #1247: [2.0] [First TF] Image context on thumbnail images

### PR DESCRIPTION
**Pull request by EricBAndrews**
# Checklist
- [x] I have read CONTRIBUTING.md
- [x] I have described what this PR contains
- [x] If this PR alters the UI, I have attached pictures/videos
- [x] This PR addresses one or more open issues that were assigned to me:
      - closes #1247 

# Pull Request Information

Adds context menu options to thumbnail images.

Also makes some minor image-related changes:
- `DynamicImageView` now has an optional `actionsEnabled` parameter to disable the context menu (e.g., for website previews)
- Various image-related helper functions have been collected into `ImageFunctions`

### CODE REVIEW

File: Mlem/App/Views/Shared/Images/Wrappers/ThumbnailImageView.swift [hunk 1]
```diff
@@ -58,6 +59,34 @@ struct ThumbnailImageView: View {
...
+                            Label(String(localized: "Share Image"), systemImage: Icons.share)
+                        }
+                        
+                        Button {
+                            Task {
+                                await showQuickLook(url: url)
+                            }
+                        } label: {
+                            Label(String(localized: "Quick Look"), systemImage: Icons.imageDetails)
+                        }
```
Comment (Sjmarf) [inline Mlem/App/Views/Shared/Images/Wrappers/ThumbnailImageView.swift]: *nit*

There's an (imo) better initializer to use here

```suggestion
                        Button("Save Image", systemImage: Icons.import) {
                            Task { await saveImage(url: url) }
                        }
                        Button("Share Image", systemImage: Icons.share) {
                            Task { await shareImage(url: url) }
                        }
                        Button("Quick Look", systemImage: Icons.imageDetails) {
                            Task { await showQuickLook(url: url) }
                        }
```

File: Mlem/App/Views/Shared/Images/Wrappers/ThumbnailImageView.swift [hunk 2]
```diff
@@ -58,6 +59,34 @@ struct ThumbnailImageView: View {
                         }
                     }
                 }
```
Comment (Sjmarf) [inline Mlem/App/Views/Shared/Images/Wrappers/ThumbnailImageView.swift:61]:
```suggestion
                }
                .contentShape(.contextMenuPreview, .rect(cornerRadius: Constants.main.smallItemCornerRadius))
```

**Review by Sjmarf — APPROVED** [review]
Looks good! A couple of minor suggestions

**Review by Sjmarf — CHANGES_REQUESTED** [review]
Hmm trying to open the post context menu always opens the image menu, even when you tap outside the image. The same as [this bug](https://github.com/mlemgroup/mlem/pull/1177#issuecomment-2266728599)

**Comment by EricBAndrews** [comment]
> Hmm trying to open the post context menu always opens the image menu, even when you tap outside the image.

Do you have a repro post? Works fine for me:

https://github.com/user-attachments/assets/70609811-d0fb-4c45-a2f7-06f1a598e757

**Comment by Sjmarf** [comment]
Happens on my iOS 18 physical device when using Compact/Headline mode. Large mode works fine. Here I'm tapping on the post itself, not the image 

https://github.com/user-attachments/assets/e510b6c1-4fb0-4525-8294-12124aa30f58

**Comment by EricBAndrews** [comment]
Interesting--I can't reproduce in the sim:

https://github.com/user-attachments/assets/9d22742a-05af-4583-a65e-e6ba4b51809a

https://github.com/user-attachments/assets/70df7264-8510-4a50-978f-a948f879c2aa

**Comment by EricBAndrews** [comment]
Ah, it's this line:

`.contentShape(.contextMenuPreview, .rect(cornerRadius: Constants.main.smallItemCornerRadius))`

Looking into a way to fix it

**Comment by EricBAndrews** [comment]
Got it! Just needed an explicit `.contentShape` as well.

**Review by Sjmarf — APPROVED** [review]

**Pull request merged**